From f73d7559c48710a195a1672a3cc2cab68022e899 Mon Sep 17 00:00:00 2001
From: Vishal Sagar <vishal.sagar@xilinx.com>
Date: Tue, 22 Sep 2020 12:24:07 -0700
Subject: [PATCH 1632/1852] v4l: xilinx-demosaic: Fix for warning UNUSED_VALUE

commit 9c98d8728b77ef86e1b2edc0ff33a01615f22c9b from
https://github.com/Xilinx/linux-xlnx.git

This patch fixes the warnings for UNUSED_VALUE.

For xdmsc_probe():
Event returned_value: 	Assigning value from
"xvip_init_resources(&xdmsc->xvip)" to "rval" here, but that stored
value is overwritten before it can be used.

Signed-off-by: Vishal Sagar <vishal.sagar@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/media/platform/xilinx/xilinx-demosaic.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/media/platform/xilinx/xilinx-demosaic.c b/drivers/media/platform/xilinx/xilinx-demosaic.c
index af464489c9e3..7bcf5ac3b624 100644
--- a/drivers/media/platform/xilinx/xilinx-demosaic.c
+++ b/drivers/media/platform/xilinx/xilinx-demosaic.c
@@ -342,6 +342,8 @@ static int xdmsc_probe(struct platform_device *pdev)
 	if (rval < 0)
 		return rval;
 	rval = xvip_init_resources(&xdmsc->xvip);
+	if (rval)
+		return -EIO;
 
 	/* Reset Demosaic IP */
 	gpiod_set_value_cansleep(xdmsc->rst_gpio,
-- 
2.31.1

